tests/unit/test_typing.py

```diff
@@ -474,3 +476,34 @@ def test_unwrap_type_alias_type_keyword() -> None:
     annotation = ctx["IntAlias"]
     field_definition = FieldDefinition.from_annotation(annotation)
     assert field_definition.is_type_alias_type
+    assert field_definition.annotation == int
+
+
+def test_unwrap_annotation_type_alias_type() -> None:
+    annotation, metadata, wrappers = unwrap_annotation(TypeAliasType("SomeAlias", int))
+    assert annotation == int
+    assert not metadata
+    assert TypeAliasType in wrappers
+
+
+NestedAlias = TypeAliasType("Outer", Union[Annotated[int, "meta"], list["Outer"]])
+
+
+@pytest.mark.parametrize(
+    "annotation, expected_meta, expected_type",
+    [
+        (Annotated[TypeAliasType("SomeAlias", int), "meta"], ("meta",), int),
+        (TypeAliasType("SomeAlias", TypeAliasType("InnerAlias", int)), (), int),
+        (TypeAliasType("SomeAlias", Annotated[int, "meta"]), ("meta",), int),
+        (TypeAliasType("SomeAlias", Annotated[TypeAliasType("InnerAlias", int), "meta"]), ("meta",), int),
+        (Annotated[TypeAliasType("SomeAlias", Annotated[int, "inner meta"]), "meta"], ("meta", "inner meta"), int),
+        (NestedAlias, ("meta",), NestedAlias),
+    ],
+)
+def test_unwrap_annotation_type_alias_type_nested(
+    annotation: Any, expected_meta: tuple[str, ...], expected_type: Any
+) -> None:
+    annotation, metadata, wrappers = unwrap_annotation(annotation)
```

Comment:
Question: is it fine that `unwrap_annotation` is not tested on `type` keyword aliases?

Reply:
Hmm. I was under the assumption that `type` always creates a `TypeAliasType`. At least that's how I understand the docs: https://docs.python.org/3.13/library/typing.html#typing.TypeAliasType

Reply:
Yes, it does. But, `TypeVarType` can be constructed manually or by the compiler. There were bugs that showed differences.